src/range/accept_ranges.rs

```diff
@@ -0,0 +1,245 @@
+use crate::headers::{HeaderName, HeaderValue, Headers, ToHeaderValues, ACCEPT_RANGES};
+
+use std::fmt::{self, Debug, Display};
+use std::option;
+
+/// HTTP Accept-Ranges
+///
+/// Accept-Ranges header indicates that the server supports
+/// range requests and specifies the unit to be used by
+/// clients for range requests.
+///
+/// The default value is to not accept range requests.
+///
+/// # Specifications
+///
+/// - [RFC 7233, section 2.3: Accept-Ranges](https://tools.ietf.org/html/rfc7233#section-2.3)
+/// - [RFC 7233, section 2.1: Byte Ranges](https://tools.ietf.org/html/rfc7233#section-2.1)
+/// - [IANA HTTP parameters, range-units: HTTP Range Unit Registry](https://www.iana.org/assignments/http-parameters/http-parameters.xhtml#range-units)
+///
+/// # Examples
+///
+/// Accepting ranges specified in byte unit (the widely used default):
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::range::AcceptRanges;
+/// use http_types::Response;
+///
+/// let accept_ranges = AcceptRanges::with_bytes();
+///
+/// let mut res = Response::new(200);
+/// accept_ranges.apply(&mut res);
+///
+/// let accept_ranges = AcceptRanges::from_headers(res)?.unwrap();
+/// assert_eq!(accept_ranges.bytes(), true);
+/// #
+/// # Ok(()) }
+/// ```
+///
+/// Other unit ranges:
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::range::AcceptRanges;
+/// use http_types::Response;
+///
+/// let custom_type = String::from("my_custom_type");
+/// let accept_ranges = AcceptRanges::with_other(custom_type.clone());
+///
+/// let mut res = Response::new(200);
+/// accept_ranges.apply(&mut res);
+///
+/// let accept_ranges = AcceptRanges::from_headers(res)?.unwrap();
+/// assert_eq!(accept_ranges.other(), &Some(custom_type));
+/// #
+/// # Ok(()) }
+/// ```
+///
+/// Range requests not accepted:
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::range::AcceptRanges;
+/// use http_types::Response;
+///
+/// let accept_ranges = AcceptRanges::new();
+///
+/// let mut res = Response::new(200);
+/// accept_ranges.apply(&mut res);
+///
+/// let accept_ranges = AcceptRanges::from_headers(res)?.unwrap();
+/// assert_eq!(accept_ranges.bytes(), false);
+/// assert_eq!(accept_ranges.other(), &None);
+/// #
+/// # Ok(()) }
+/// ```
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct AcceptRanges {
+    bytes: bool,
+    other: Option<String>,
+}
+
+impl AcceptRanges {
+    const BYTES: &'static str = "bytes";
+    const NONE: &'static str = "none";
+
+    /// Create a new AcceptRange which does not accept range requests.
+    pub fn new() -> Self {
```

Comment:
`new` setting the value to `"none"` might be confusing here since the widely used default is `"bytes"`. Maybe we should name it `new_none` or a better  idea ?